[DOC] Remove --stability-level for Alloy v1.2

knylander-grafana opened the pull request:
Changes the command line for starting Alloy to not use --stability-level when starting Alloy v1.2. 

This PR should go live when Alloy 1.2 is released.

knylander-grafana commented on docs/sources/configure-client/grafana-agent/java/_index.md:
```diff
@@ -78,13 +78,27 @@ and `discover.process` components to work.
+
+The `stability.level` option is required for `pyroscope.scrape` with Alloy v1.0 or v1.1. For more information about `stability.level`, refer to [The run command](https://grafana.com/docs/alloy/latest/reference/cli/run/#permitted-stability-levels) documentation.
+
+To start Grafana Alloy, replace `configuration.alloy` with your configuration filename:
 
 `alloy run --stability.level=public-preview configuration.alloy`
 
 The `stability.level` option is required for `pyroscope.scrape`. For more information about `stability.level`, refer to [The run command](https://grafana.com/docs/alloy/latest/reference/cli/run/#permitted-stability-levels) documentation.
```
```suggestion
```

clayton-cornell reviewed (approved):
Looks OK to me. Everything lines up :-)

korniltsev reviewed (approved):
lgtm, thanks \m/

knylander-grafana commented:
> Line 12-14 in `docs/sources/configure-client/grafana-agent/java/_index.md` should probably be updated as well. River only applies to Agent Flow. Alloy uses [grafana.com/docs/alloy/latest/concepts/configuration-syntax](https://grafana.com/docs/alloy/latest/concepts/configuration-syntax/) (which is the same as River, but it's not called River in Alloy.

I've updated the text and links. Let me know if what I added is correct.

clayton-cornell commented on docs/sources/configure-client/grafana-agent/java/_index.md:
```diff
@@ -9,9 +9,11 @@ weight: 20
+
+The collector configuration file is composed of components that are used to collect,
 transform, and send data.
+Alloy configuration files use [configuration syntax](https://grafana.com/docs/alloy/latest/concepts/configuration-syntax/).
```
```suggestion
Alloy configuration files use the Alloy [configuration syntax](https://grafana.com/docs/alloy/latest/concepts/configuration-syntax/).
```
